### ISSUE TITLE
[HW] Create software architecture overview

### ISSUE THREAD

**Issue by kuoirene** [amosproj/amos2024ss05-knowledge-graph-extractor#7]
## User story
1. As a dev team
2. I need to create an overview of the current software architecture
3. So that there is a high-level overview of the structure of the product and how all parts are interconnected

## Acceptance criteria
* An overview diagram of runtime components (one page)
* An overview diagram of code components (one page)
* A summary of the underlying technology stack (one page)
* A textual explanation of the diagrams and choices (one page)

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
